bgp/__init__.py

```diff
@@ -182,6 +176,16 @@ def sequence(self, book):
             raise Exception(sq.book.identifier + ' - DjvuXML and/or DjvuTXT is forbidden and can\'t be sequenced!')
             logging.error(sq.book.identifier + ' - DjvuXML and/or DjvuTXT is forbidden and can\'t be sequenced!')
 
+    @classmethod
+    def _upload(cls, itemid=None, results=None):
+        itemid = itemid or results['identifier']
```

Comment:
No matter what there should be results, right? Either we pass in results, or the results is on the object.

Therefore, we likely don't need an `itemid`; it should always be the case that `results['identifier']` is set.

If so, we can likely remove `itemid` from `_upload`'s declaration.

I'd also likely change L181 to be `itemid = results.get('identifier')` and if there isn't an `itemid`, log and raise an exception (which gets handled by the pipeline)